forte/data/data_store.py

```diff
@@ -1203,21 +1205,91 @@ def co_iterator_annotation_like(
 
         The precedence of those values indicates their priority in the min heap
         ordering.
-        For example, if two entries have both the same begin and end field,
-        then their order is
+
+        Lastly, the `range_begin` and `range_end` argument determines the start
+        and end position of the span range within which entries of specified by
+        `type_name` need to be fetched. For example, if two entries have both
+        the same begin and end field, then their order is
         decided by the order of user input type_name (the type that first
         appears in the target type list will return first).
         For entries that have the exact same `begin`, `end` and `type_name`,
         the order will be determined arbitrarily.
 
+        For example, let's say we have two entry types,
+        :class:`~ft.onto.base_ontology.Sentence` and
+        :class:`~ft.onto.base_ontology.EntityMention`.
+        Each type has two entries. The two entries of type `Sentence` ranges from span
+        `(0,5)` and `(6,10)`. Similarly, the two entries of type `EntityMention` has span
+        `(0,3)` and `(15,20)`.
+
+        .. code-block:: python
+
+            # function signature
+            entries = list(
+                co_iterator_annotation_like(
+                    type_names = [
+                        "ft.onto.base_ontology.Sentence",
+                        "ft.onto.base_ontology.EntityMention"
+                    ],
+                    range_begin = 0,
+                    range_end = 12
+                )
+            )
+
+            # Fetching result
+            result = [
+                all_anno.append([type(anno).__name__, anno.begin, anno.end])
+                for all_anno in entries
+            ]
+
+            # return
+            result = [
+                ['Sentence', 0, 5],
+                ['EntityMention', 0, 5],
+                ['Sentence', 6, 10]
+            ]
+
+        From this we can see how `range_begin` and `range_end` affect which
+        entries will be fetched and also how the function chooses the order
+        in which entries are fetched.
+
         Args:
             type_names: a list of string type names
+            range_begin: start index of the range in
+                which we want to get required entries
+            range_end: end index of the range in
+                which we want to get required entries
 
         Returns:
             An iterator of entry elements.
         """
 
-        n = len(type_names)
+        def get_bisect_range(
```

Comment:
maybe we should make it a class method so that other class methods can use it as well